atlasdb-commons/src/test/java/com/palantir/common/base/BatchingVisitablesTest.java

```diff
@@ -0,0 +1,416 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.common.base;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+import org.junit.Test;
+
+import com.google.common.base.Function;
+import com.google.common.base.Functions;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Predicate;
+import com.google.common.base.Predicates;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Ordering;
+import com.google.common.collect.UnmodifiableIterator;
+import com.palantir.util.Mutable;
+import com.palantir.util.Mutables;
+import com.palantir.util.Pair;
+import com.palantir.util.paging.TokenBackedBasicResultsPage;
+
+@SuppressWarnings("Guava") // BatchingVisitables uses Guava.
```

Comment:
do we need this?

Reply:
Not for correct compilation. However, I like to avoid having warnings littering the code, as it disguises new warnings which can be useful to see. In this case, we get warnings in the IDE advising us to use native Java instead of Guava; however, the source of the issue is in BatchingVisitables, and I didn't want to switch those over in this PR.